tests/test_pytest/test_dtree.py

```diff
@@ -0,0 +1,43 @@
+# # # Distribution Statement A. Approved for public release. Distribution unlimited.
+# # #
+# # # Author:
+# # # Naval Research Laboratory, Marine Meteorology Division
+# # #
+# # # This program is free software: you can redistribute it and/or modify it under
+# # # the terms of the NRLMMD License included with this program. This program is
+# # # distributed WITHOUT ANY WARRANTY; without even the implied warranty of
+# # # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the included license
+# # # for more details. If you did not receive the license, for more information see:
+# # # https://github.com/U-S-NRL-Marine-Meteorology-Division/
+
+"""Takes in a dictionary of xarrays and converts to xarray datatree"""
+
+
+import logging
+import pytest
+from datatree import DataTree
+import glob
+from geoips.interfaces import readers
+    
+LOG = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def load_testfiles():
+    """Preload files for testing"""
+
+    #only want a few files for testing
+    fnames = glob.glob('geoips/test_data/test_data_amsr2/data/*.nc')[:2]
+    amsr2_reader = readers.get_plugin('amsr2_netcdf')
+    xarray_dict = amsr2_reader(fnames)
+    return xarray_dict
+
+
+def test_xarray_to_datatree(load_testfiles):
+    """Convert a xarray dictionary to a datatree"""
+
+    #check that the datatree is equal to the dict-xarray
+    xarray_datatree = DataTree.from_dict(load_testfiles)
+    assert xarray_datatree == load_testfiles

```

Comment:
should we call the xr_to_dtree function here?  And do we want to ensure the structure of the resulting dtree is "correct"?

We also might want to think about using a different file than AMSR2, AMSR2 is probably not the fastest one to read (I expect we want these unit tests to run fairly quickly)

Reply:
Yes, unit tests are intended to test the actual GeoIPS code so this should be testing `xr_to_dtree`.

How long does this test take to run?

Generally, though, I think it would be better to avoid using any actual test data in the unit tests. They should be able to run without actual test data. Is there a good way for us to set up a dummy xarray dictionary directly in this script to test against?

Reply:
@AJThorpe can we create a dummy geoips-like dictionary of xarrays directly in this unit test, then pass it to xr_to_dtree, then make sure the output is what we expect?

Reply:
Oh,.  You already did it. :-)